app/Http/Controllers/UserController.php

```diff
@@ -2,50 +2,191 @@
 
 namespace Suyabay\Http\Controllers;
 
+use Validator;
+use Hash;
 use Suyabay\User;
 use Suyabay\Role;
 use Suyabay\Invite;
 use Suyabay\Http\Requests;
 use Illuminate\Http\Request;
-use Illuminate\Http\Response;
+use Illuminate\Support\Facades\Response;
 use Illuminate\Database\QueryException;
 use Suyabay\Http\Controllers\Controller;
+use League\Fractal\Manager;
+use League\Fractal\Resource\Collection;
+use Suyabay\Http\Transformers\UserTransformer;
 
 class UserController extends Controller
 {
     protected $mail;
     protected $response;
+    protected $fractal;
+
+    public function __construct(Manager $fractal)
+    {
+        $this->fractal = $fractal;
+    }
 
     /**
      * This method return all users to the calling API endpoint
      */
-    public function getAllUsers(Response $response)
+    public function getAllUsers(Request $request)
+    {
+        $perPage = 3;
+
+        $page = $request->query('page') ? : 1;
+
+        $totalPage = (int) ($perPage*$page)-$perPage;
+
+        $users = User::orderBy('id', 'asc')
+        ->skip($totalPage)
+        ->take($perPage)
+        ->get([
+            'id',
+            'username',
+            'email',
+            'created_at',
+            'updated_at',
+            'avatar'
+        ]);
+
+        $resource = new Collection($users, new UserTransformer());
+
+        if (isset($resource)) {
+            $data = $this->fractal->createData($resource)->toArray();
+
+            return Response::json($data, 200);
+
+        }
+
+        return Response::json(['message' => 'Users Not found'], 404);
+        
+    }
+
+    /**
+     * This method return a single user to the calling API endpoint getMyDetails
+     */
+    public function getSingleUser($username)
     {
-        $users = User::orderBy('id', 'asc')->get(['id', 'username', 'email', 'created_at', 'updated_at', 'avatar']);
+        $user = User::where('username', '=', $username)->get([
+            'id',
+            'username',
+            'email',
+            'created_at',
+            'updated_at',
+            'avatar'
+        ]);
+
+        $resource = new Collection($user, new UserTransformer());
```

Comment:
**Code Quality - Critical**
@andela-tolotin Since you know you are returning a single model, you should make use of  Fractal Item and not Collection